sdk/keyvault/azure-keyvault-secrets/tests/_async_test_case.py

```diff
@@ -0,0 +1,48 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+import pytest
+from azure.keyvault.secrets._shared.client_base import DEFAULT_VERSION
+from devtools_testutils import AzureRecordedTestCase, is_live
+from azure_devtools.scenario_tests.exceptions import AzureTestError
+
+
+class AsyncSecretsClientPreparer(AzureRecordedTestCase):
+    def __init__(self, **kwargs) -> None:
+        self.azure_keyvault_url = (
+            os.getenv("AZURE_KEYVAULT_URL", None) if is_live() else "https://vaultname.vault.azure.net"
+        )
+
+        if not self.azure_keyvault_url:
```

Comment:
Sorry for flip-flopping on this, but I just realized that we already do this type of handling in `azure-keyvault-keys`: [link](https://github.com/Azure/azure-sdk-for-python/blob/c39db1cc0d9b4a97cce086b8d040acdfcbda6988/sdk/keyvault/azure-keyvault-keys/tests/_test_case.py#L107-L109). We should probably do the same thing here that we do there; we can implicitly require the `AZURE_KEYVAULT_URL` variable by directly accessing `os.env["AZURE_KEYVAULT_URL"]` since that will raise a pretty obvious key error if it fails